File: Data Structures & Algorithms/two-integer-sum/submission-5.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hm = new HashMap<>();
        for(int i = 0; i<nums.length; i++){
            hm.put(nums[i],i);
        }
        for(int i = 0; i<nums.length; i++){
            int difference = target - nums[i];
            if(hm.containsKey(difference) && i!=hm.get(difference)){
                 return new int[] {i,hm.get(difference)};
            }
        }
        return new int[] {0,0} ;
    }
}

File: Data Structures & Algorithms/house-robber/submission-5.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = [0] * (len(nums)+1)
        
        for i in range(len(nums)-1, -1, -1):
            if i == len(nums) - 1:
                dp[i] = nums[i]
            elif(i == len(nums) - 2):
                dp[i] = max(nums[i], dp[i+1])
            else:
                dp[i] = max(nums[i] + dp[i+2], dp[i+1])
        return dp[0]
